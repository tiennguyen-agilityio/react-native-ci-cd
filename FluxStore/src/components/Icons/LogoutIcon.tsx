import {memo} from 'react';
import {TouchableOpacity} from 'react-native';
import Svg, {Path} from 'react-native-svg';

import {IconProps} from '@/interfaces';
import {useThemeStore} from '@/stores';

export const LogoutIcon = memo(
  ({width = 21, height = 21, color, disabled = false, style, onPress}: IconProps) => {
    const {
      theme: {icon},
    } = useThemeStore();

    return (
      <TouchableOpacity
        testID="logout-icon"
        activeOpacity={0.8}
        onPress={onPress}
        disabled={disabled}
        style={style}>
        <Svg width={width} height={height} viewBox="0 0 21 21" fill="none">
          <Path
            fillRule="evenodd"
            clipRule="evenodd"
            d="M7.658 8.826a.674.674 0 00-.685.674.68.68 0 00.685.674H13v4.182c0 2.144-1.772 3.894-3.962 3.894H4.703C2.522 18.25.75 16.509.75 14.365V4.644C.75 2.49 2.53.75 4.712.75h4.344C11.228.75 13 2.491 13 4.635v4.191H7.658zm8.518-2.353l2.555 2.546a.668.668 0 010 .954l-2.555 2.546a.678.678 0 01-.472.201.676.676 0 01-.481-1.155l1.4-1.391H13V8.826h3.623l-1.4-1.39a.676.676 0 010-.955.668.668 0 01.953-.008z"
            fill={color ?? icon.default}
          />
        </Svg>
      </TouchableOpacity>
    );
  },
);
