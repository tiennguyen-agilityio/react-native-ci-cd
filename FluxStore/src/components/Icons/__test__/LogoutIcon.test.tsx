import {render, fireEvent, screen} from '@testing-library/react-native';

import {LogoutIcon} from '../LogoutIcon';

describe('LogoutIcon', () => {
  const onPress = jest.fn();

  it('renders correctly (matches snapshot)', () => {
    const {toJSON} = render(<LogoutIcon />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('calls onPress when touched and not disabled', () => {
    render(<LogoutIcon onPress={onPress} />);

    fireEvent.press(screen.getByTestId('logout-icon'));

    expect(onPress).toHaveBeenCalledTimes(1);
  });

  it('does NOT call onPress when disabled', () => {
    render(<LogoutIcon onPress={onPress} disabled />);

    fireEvent.press(screen.getByTestId('logout-icon'));

    expect(onPress).not.toHaveBeenCalledTimes(0);
  });
});
