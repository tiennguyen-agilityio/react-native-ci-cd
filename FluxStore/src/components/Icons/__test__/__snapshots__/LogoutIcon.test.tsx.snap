// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoutIcon renders correctly (matches snapshot) 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="logout-icon"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={21}
    bbWidth={21}
    fill="none"
    focusable={false}
    height={21}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 21,
          "width": 21,
        },
      ]
    }
    vbHeight={21}
    vbWidth={21}
    width={21}
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        clipRule={0}
        d="M7.658 8.826a.674.674 0 00-.685.674.68.68 0 00.685.674H13v4.182c0 2.144-1.772 3.894-3.962 3.894H4.703C2.522 18.25.75 16.509.75 14.365V4.644C.75 2.49 2.53.75 4.712.75h4.344C11.228.75 13 2.491 13 4.635v4.191H7.658zm8.518-2.353l2.555 2.546a.668.668 0 010 .954l-2.555 2.546a.678.678 0 01-.472.201.676.676 0 01-.481-1.155l1.4-1.391H13V8.826h3.623l-1.4-1.39a.676.676 0 010-.955.668.668 0 01.953-.008z"
        fill={
          {
            "payload": 2147483648,
            "type": 0,
          }
        }
        fillRule={0}
        propList={
          [
            "fill",
            "fillRule",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
